### PR TITLE
Add `auto-toggle` command that turns light on/off when using a webcam

### PR DESCRIPTION
**Pull request by Holzhaus**
This adds a new `auto-toggle` command to the command line interface. When using that command, `litra` will keep running and monitor the video device files (`/dev/video*`) using inotify. Opening or closing such a device will be tracked using a counter to determine if at least one device is currently opened. If so, the light is turned on, otherwise it's turned off.

*Note:* This is a quick & dirty PoC implementation and will only works on Linux. I suppose it can be improved, but I don't know if and when I find the time to do that. It works for me for now :)

### CODE REVIEW

**Comment by Holzhaus** [comment]
> @Holzhaus If you don't mind, I think I might hold off on this in hope of a cross-platform solution. We now have a Cargo crate that exposes a library target, so maybe this could be a separate OSS project? 👼🏻 



The feature is opt-in, so it wouldn't break anything in case you worry about that (without it, the `auto-toggle` subcommand simply doesn't exist). But I can also make it a separate project if you want.

**Comment by timrogers** [comment]
> The feature is opt-in, so it wouldn't break anything in case you worry about that (when building without the feature flag, the `auto-toggle` subcommand simply doesn't exist). But I can also make it a separate project if you want.

That makes sense! I'm not familiar with Rust's support for features, so I didn't know you could do that 👼🏻 

What do you think of merging this change, and then updating our CI/CD process for creating binaries to build with `--features autotoggle` when building for Linux?

**Comment by timrogers** [comment]
@Holzhaus Is this done from your perspective? 

Does it make sense for me to update the CI workflows to include the new feature when building for Linux?

**Comment by Holzhaus** [comment]
> @Holzhaus Is this done from your perspective?

Yes.

> Does it make sense for me to update the CI workflows to include the new feature when building for Linux?

With https://github.com/timrogers/litra-rs/pull/18/commits/b82b4bde2f42cee1f6bd889ab372ef8f509798e6 it should be automatically included on Linux, and skipped on other OSes.

If you want to add support for other OSes: You basically need some kind of monitoring for accesses to the camera device(s). No idea how this works on Windows. However, the Logitech G HUB application has a similar feature, so windows users can use that for now if they really need it.

**Comment by timrogers** [comment]
@Holzhaus This looks good to me! Let's get #21 merged, and then we can rebase this and get it shipped too.

**Review by timrogers — APPROVED** [review]

**Comment by timrogers** [comment]
@Holzhaus Would you be able to rebase this one, and then I'll get it merged in?

**Comment by Holzhaus** [comment]
> @Holzhaus Would you be able to rebase this one, and then I'll get it merged in?

Done.

File: src/main.rs [hunk 1]
```diff
@@ -67,6 +69,25 @@ enum Commands {
         #[clap(long, short, action, help = "Return the results in JSON format")]
         json: bool,
     },
+    #[cfg(target_os = "linux")]
+    /// Automatically switch of the device on and off if the webcam is being used.
```
Comment (timrogers) [inline src/main.rs:73]:
```suggestion
    /// Automatically turn the Logitech Litra device on when your webcam turns on, and off when the webcam turns off
```

**Review by timrogers — APPROVED** [review]

**Pull request merged**